src/main/java/org/testng/internal/BaseTestMethod.java

```diff
@@ -763,9 +763,15 @@ private IRetryAnalyzer getRetryAnalyzerConsideringMethodParameters(ITestResult t
     boolean relyOnParametersToBeUsedAsKeys = relyOnParametersToBeUsedAsKeys(key);
     IRetryAnalyzer retryAnalyzer = this.m_retryAnalyzer;
     if (key.length != 0 && retryAnalyzer != null && relyOnParametersToBeUsedAsKeys) {
-      String keyAsString = getMethodName() + "_" + Arrays.toString(key);
-      retryAnalyzer = m_testMethodToRetryAnalyzer.computeIfAbsent(keyAsString,
-          o -> InstanceCreator.newInstance(this.m_retryAnalyzer.getClass()));
+      final String keyAsString = getMethodName() + "_" + Arrays.toString(key);
+      final IRetryAnalyzer currentRetryAnalyzerInMap = m_testMethodToRetryAnalyzer.get(keyAsString);
+      if (currentRetryAnalyzerInMap == null || currentRetryAnalyzerInMap.getClass() != retryAnalyzer.getClass()) {
+        retryAnalyzer = m_testMethodToRetryAnalyzer.compute(

```

Comment:
I think you don't need to compute as you already get the retry-analyzer and know you want to create a new one.
Maybe `retryAnalyzer = InstanceCreator.newInstance(m_retryAnalyzer.getClass()))` will be enough

Reply:
> I think you don't need to compute as you already get the retry-analyzer and know you want to create a new one.
> Maybe `retryAnalyzer = InstanceCreator.newInstance(m_retryAnalyzer.getClass()))` will be enough

Won't be enough because it is not added to the map. To use `put` is another option here.

Reply:
As you prefer. It's a detail but I think `compute` is strange here as you don't use parameters. `put` looks better.

Reply:
Should I change it?